packages/core/src/upload-manager.ts

```diff
@@ -70,6 +71,62 @@ class UploadManager {
       throw new Error(errorMessage);
     }
   }
+  public async pinCID(configuration: {
+    name: string;
+    organizationId?: string;
+    token: string;
+    cid: string;
+  }): Promise<{
+    deploymentId: string;
+    projectId: string;
+    sitePreview: string;
+    affectedDomains: string[];
+  }> {
+    try {
+      if (!configuration.name) {
+        throw new Error("Name is not provided.");
```

Comment:
```suggestion
        throw new Error("Bucket name is not provided.");
```
Do this on other methods which validate this